shotgun_api3/shotgun.py

```diff
@@ -3853,6 +3851,23 @@ def _complete_multipart_upload(self, upload_info, filename, etags):
         if not str(result).startswith("1"):
             raise ShotgunError("Unable get upload part link: %s" % result)
 
+    def _requires_direct_s3_upload(self, entity_type, field_name):
+        """
+        Internal function that determines if an entity_type + field_name combination
+        should be uploaded to cloud storage.
+
+        :param str entity_type: The entity type of the file being uploaded.
+        :param str field_name: The matching field name for the file being uploaded.
+
+        :returns: Whether the field + entity type combination should be uploaded to cloud storage.
+        :rtype: bool
+        """
+        supported_s3_types = self.server_info.get('s3_enabled_upload_types', {})
+        supported_fields = supported_s3_types.get(entity_type, [])
+        return self.server_info.get("s3_direct_uploads_enabled", False) and \
+            (field_name in supported_fields or field_name in supported_s3_types.get("*", []) or
+                (isinstance(supported_fields, list) and "*" in supported_fields) or supported_fields == "*")
```

Comment:
do we need `isinstance(supported_fields, list)`? Isn't it always a list?

Reply:
The one case I figured it wouldn't be a list is if star is the value. For example I think it makes sense for `s3_enabled_upload_types` to be:
```json
{ "Version": "*" }
```

I didn't want to make too many assumptions about what the api returned. Would it make more sense to change it to only lists to be consistent?